samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -70,18 +70,33 @@ def normalize(template_dict, normalize_parameters=False):
                     },
                 )
 
+        # This is a work around to allow the customer to use sam deploy or package commands without the need to provide
+        # values for the CDK auto generated asset parameters. The suggested solution is to let CDK add some metadata to
+        # the autogenerated parameters, so sam can skip them safely.
+
+        # Normalizing CDK auto generated parameters. Search for parameters that meet these conditions:
+        # 1- parameter name starts with AssetParameters
+        # 2- parameter type is string
+        # 3- there is no reference to this parameter any where in the template resources
+        # 4- there is no default value for this parameter
```

Comment:
Can we add additional checks for these parameters? I remember they have random chars after `AssetParameters`, can we create a pattern and check for that?

Reply:
done